src/pipx/constants.py

```diff
@@ -49,16 +53,24 @@ def is_mingw() -> bool:
     eval "$(register-python-argcomplete pipx)"
 
 zsh:
-    To activate completions for zsh you need to have
-    bashcompinit enabled in zsh:
+    To activate completions in zsh, first make sure compinit is marked for
+    autoload and run autoload:
 
-    autoload -U bashcompinit
-    bashcompinit
+    autoload -U compinit && compinit
 
-    Afterwards you can enable completion for pipx:
+    Afterwards you can enable completions for pipx:
 
     eval "$(register-python-argcomplete pipx)"
 
+    > [!NOTE]
+    >
+    > If your version of argcomplete is earlier than v3, you may need to
+    have bashcompinit enabled in zsh by running:
+    >
+    > autoload -U bashcompinit
+    > bashcompinit
```

Comment:
Sorry, I must have thought this documentation was inside a markdown file, for a moment (well, for more than a moment, since creating GH suggestions with the right indentation is a pain)!

```suggestion
    NOTE: If your version of argcomplete is earlier than v3, you may need to
    have bashcompinit enabled in zsh by running:
    
    autoload -U bashcompinit
    bashcompinit
```

Reply:
Haha, no worries! I certainly had to correct some markdown notation when I was first writing it, too 😂